src/Civi/Buildkit/Phars.php

```diff
@@ -63,6 +71,12 @@ protected static function findUpdates(array $phars): array {
    * @throws \Exception
    */
   protected static function downloadFile(string $remoteUrl, string $expectHash, string $localFile, int $blockSize = 65536): void {
+    if (is_dir($localFile)) {
+      static::printf("    WARNING: Cannot overwrite folder (%s) with file. Skip download.\n", $localFile);
+      // This should arguably be fatal... I'm not sure it's going to bubble-up
```

Comment:
Fixed the issue of making the warning "bubble up" to console -- that's the change to `composer.json` (`compile-passthru=>always`)